src/middlewares/errorMiddleware.js

```diff
@@ -0,0 +1,23 @@
+import { ApiError } from '../exeptions/api.error.js';
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
+  // eslint-disable-next-line no-console
+  console.error(error);
+
+  if (error) {
+    res.statusCode = 500;
+
+    res.send({
+      message: 'Server error',
+    });
+  }
```

Comment:
The 'next()' function should not be called after sending a response to the client. Once a response is sent, the middleware chain should be terminated to prevent further processing.